CIPs/0007.md

```diff
@@ -0,0 +1,108 @@
+# CIP [0007]: Affordable Gas Downtime Slasher
+
+- Date: 2020-05-05
+- Author: @gastonponti (with a lot of feedback and ideas from @m-chrzan)
+- Status: WIP
+
+## Overview
+
+This proposal contains a new DowntimeSlasher contract.
+
+The actual DowntimeSlasher tries to verify that the validator signed at least a block in all the window that is required to be validated which it leads to running out of gas for big windows.
```

Comment:
We would like any readers unfamiliar with the issue that prompted this proposal, and who may be external to cLabs, to have the context to understand and provide comments on the proposal.
```suggestion
Currently, slashable downtime is defined as a window of blocks, greater than a predefined length, in which no block's parent aggregated seal includes a signature from the validator in question. It's currently implementation checks and executes slashing in a single transaction which requires an amount of gas linear in the required window size.

Under its current implementation, the gas required to evaluate a downtime event may far exceed the block gas limit. In the Release Candidate 1 network, the downtime slashing window is configured to be 5760 blocks, which would require roughly 13.5 million gas to evaluate, while the block gas limit is set to 10 million.
```